src/GraphClient.ts

```diff
@@ -50,25 +55,68 @@ export default class GraphClient {
     // GraphQL query that is parsed into the standard GraphQL AST(Abstract syntax tree)
     const gqlSubscriptionQry = gql`${subscriptionQry}`;
     // Subscription handling
-    const querySubscriber = await Promise.resolve(this.apolloClient.subscribe({
+    const observable = this.apolloClient.subscribe({
       query: gqlSubscriptionQry,
       variables: {},
-    }).subscribe({
-      async next(response) {
-        const transactions = await fetcher.fetch(response.data);
-        await handler.handle(transactions);
-        // Integrate updation of ContractEntity uts here. Handlers should make sure error
-        // Promise.reject is thrown on error cases.
-      },
-      error(err) {
-        // Log error using logger
-        Logger.error(err);
-      },
-    }));
+    });
+    const querySubscriber = await Promise.resolve(
+      observable
+        .subscribe({
+          async next(response) {
+            const transactions = await fetcher.fetch(response.data);
+            await handler.handle(transactions);
+            await GraphClient.updateLatestUTS(
+              transactions,
+              response.data,
+              contractEntityRepository,
+            );
+          },
+          error(err) {
+            Logger.error(err);
+          },
+        }),
+    );
 
     return querySubscriber;
   }
 
+  /**
+   * This method updates latest timestamp for contract entities.
+   * @param transactions Transactions for transaction fetcher.
+   * @param subscriptionResponse Subscription response.
+   * @param contractEntityRepository Instance of contract entity repository.
+   */
+  private static async updateLatestUTS(
+    transactions: Record<string, Record<string, any>>,
+    subscriptionResponse: Record<string, any[]>,
+    contractEntityRepository: ContractEntityRepository,
+  ): Promise<void> {
+    const savePromises = Object.keys(transactions).map(
+      async (transactionKind) => {
+        const { contractAddress } = subscriptionResponse[transactionKind][0];
```

Comment:
Can it be that subscriptionResponse[transactionKind] is empty? Then indexing 0 value will crash.

Reply:
I guess no Because subscription call back is triggered when there is some data.